src/binutilplusplus.ts

```diff
@@ -0,0 +1,371 @@
+        } else {
+            const found = await context.fs.existsAsync(configuredPath);
+            return { found, how: 'config', where: configuredPath };
+        }
+    }
+
+    // No config so look on the system PATH
+    const cmd = context.shell.isWindows() ? `where.exe ${context.binary.binBaseName}.exe` : `which ${context.binary.binBaseName}`;
+
+    const opts = {
+        async: true,
+        env: {
+            HOME: process.env.HOME,
+            PATH: process.env.PATH
+        }
+    };
+
+    const execResult = await context.shell.execCore(cmd, opts);
+    if (execResult.code !== 0) {
+        return { found: false, how: 'path' };
+    }
+
+    return { found: true, how: 'path', where: execResult.stdout };
+}
+
+// This is silent - building block for experiences that need an input->output invoke
+export async function invokeForResult(context: Context, command: string, stdin: string | undefined): Promise<ExecResult> {
+    const fbr = await findBinary(context);
+    if (!fbr.found) {
+        return { resultKind: 'exec-bin-not-found', execProgram: context.binary, command, findResult: fbr };
+    }
+
+    const bin = await baseBinPath(context);
+    const cmd = `${bin} ${command}`;
+    const sr = await context.shell.exec(cmd, stdin);
+
+    if (!sr) {
+        return { resultKind: 'exec-failed', execProgram: context.binary, command };
+    }
+
+    if (sr.code === 0) {
+        return { resultKind: 'exec-succeeded', execProgram: context.binary, command, stdout: sr.stdout };
+    }
+
+    return { resultKind: 'exec-errored', execProgram: context.binary, command, code: sr.code, stderr: sr.stderr };
+}
+
+// TODO: can probably get rid of this now though would be nice
+// to have a kill feature
+export interface RunningProcess {
+    readonly lines: rx.Observable<string>;
+}
+
+export async function invokeTracking(context: Context, args: string[]): Promise<RunningProcess> {
+    const linesSubject = new rx.Subject<string>();
+
+    const fbr = await findBinary(context);
+    if (!fbr.found) {
+        linesSubject.error({ resultKind: 'exec-bin-not-found', execProgram: context.binary, command: args.join(' '), findResult: fbr });
+    }
+
+    const bin = await baseBinPath(context);
+
+    let pending = '';
+    const stdout = spawnrx.spawn(bin, args);
+    stdout.subscribe((chunk) => {
+        const todo = pending + chunk;
+        const lines = todo.split('\n').map((l) => l.trim());
+        const lastIsWholeLine = todo.endsWith('\n');
+        pending = lastIsWholeLine ? '' : lines.pop()!;
+        for (const line of lines) {
+            linesSubject.next(line);
+        }
+    });
+    return { lines: linesSubject };
+}
+
+export async function invokeBackground(context: Context, args: string[]): Promise<BackgroundExecResult> {
+    const fbr = await findBinary(context);
+    if (!fbr.found) {
+        return { resultKind: 'exec-bin-not-found', execProgram: context.binary, command: args.join(' '), findResult: fbr };
```

Comment:
Wouldn't it be better to have constants instead of writing strings every time? (exec-failed, exec-errorer, exec-succeeded...)

Reply:
I did this for something else on another branch and didn't feel it was terribly worth the effort - because these are not general-purpose strings but statically checked instances of the `exec-bin-not-found` (or whatever) _type_, you get compiler checking anyway.  But yeah, would be easier to change if they were defined as consts of the relevant type.